### PR TITLE
#129 clarify timeout plus some minor tidy up

### PR DESCRIPTION
**Pull request by Emily-Jiang**
Signed-off-by: Emily Jiang <emijiang6@googlemail.com>

### CODE REVIEW

File: spec/src/main/asciidoc/timeout.asciidoc [hunk 1]
```diff
@@ -42,7 +42,16 @@ The above code-snippet means the method serviceA applies the `Timeout` policy,
 which is to fail the execution if the execution takes more than 400ms to complete even if it successfully returns.
 
 When a timeout occurs, A `TimeoutException` must be thrown.
-The `Timeout` annotation can be used together with `Fallback`, `CircuitBreaker`, `Asynchronous`, `Bulkhead` and `Retry`.
-A `Fallback` can be specified and it will be invoked if the `TimeoutException` is thrown.
-If `Timeout` is used together with `Retry`, the `TimoutException` will trigger the retry.
-When `Timeout` is used with `CircuitBreaker` and if a `TimeoutException` occurs, the failure will contribute towards the circuit open.
+The `@Timeout` annotation can be used together with `@Fallback`, `@CircuitBreaker`, `@Asynchronous`, `@Bulkhead` and `@Retry`.
+
+When `@Timeout` is used without `@Asynchronous`, on reaching the specified timeout duration, attempt will be made to interrupt the operation thread. The interruption will only work in certain scenarios. The interruption will not work for the following situations:
```
Comment (OndrejM) [inline spec/src/main/asciidoc/timeout.asciidoc]: This reads better:

> When `@Timeout` is used without `@Asynchronous`, the current thread will be interrupted with a call to `Thread.interrupt()` on reaching the specified timeout duration.

File: spec/src/main/asciidoc/timeout.asciidoc [hunk 2]
```diff
@@ -42,7 +42,16 @@ The above code-snippet means the method serviceA applies the `Timeout` policy,
...
+
+* The thread is blocked on blocking I/O (database, file read/write), an exception is thrown only in case of waiting for a NIO channel
+* The thread isn't waiting (CPU intensive task) and isn't checking for being interrupted
+* The thread will catch the interrupted exception (with a general catch block) and will just continue processing, ignoring the interrupt
```
Comment (OndrejM) [inline spec/src/main/asciidoc/timeout.asciidoc]: I would delete the comment "(with a general catch block)" - it's just for illustration and unnecessary in a formal document

File: spec/src/main/asciidoc/timeout.asciidoc [hunk 3]
```diff
@@ -42,7 +42,16 @@ The above code-snippet means the method serviceA applies the `Timeout` policy,
...
+
+In the above situations, it is impossible to suspend the execution. The execution thread will finish its process. If the execution takes longer than the specified timeout, the `TimeoutException` will be thrown and the execution result will be discard.
```
Comment (OndrejM) [inline spec/src/main/asciidoc/timeout.asciidoc]: A typo:
... will be discard**ed**.

File: spec/src/main/asciidoc/timeout.asciidoc (hunk 1 above)
Comment (Emily-Jiang) [reply spec/src/main/asciidoc/timeout.asciidoc]: done

File: spec/src/main/asciidoc/timeout.asciidoc (hunk 2 above)
Comment (Emily-Jiang) [reply spec/src/main/asciidoc/timeout.asciidoc]: done

File: spec/src/main/asciidoc/timeout.asciidoc (hunk 3 above)
Comment (Emily-Jiang) [reply spec/src/main/asciidoc/timeout.asciidoc]: done

**Comment by Emily-Jiang** [comment]
@johnament it can. The only thing is that if the thread is not interruptible, the timeout exception will be delayed.

**Comment by OndrejM** [comment]
@johnament, it's explained here:

> In the above situations, it is impossible to suspend the execution. The execution thread will finish its process. If the execution takes longer than the specified timeout, the `TimeoutException` will be thrown and the execution result will be discarded.

It means everything will work, but the exception will be delayed until the annotated method finishes. Feel free to suggest a better explanation.

**Pull request merged**